### DISCUSSION TITLE
how user is inserted in socket.request.user

### DISCUSSION THREAD

**Discussion by harshilsharmaa** [socketio/socket.io#4403] [Q&A]
how it is getting user from socket.request.user because in code it is not updated when user login.
I want to authenticate if user is logged in only then the socketio will connect.

**Comment by darrachequesne** [comment]
Hi!

The `socket.request.user` is inserted during the handshake (the middleware reads the cookies from the `request` object).

Please check the following examples:

- https://socket.io/how-to/use-with-express-session
- https://github.com/socketio/socket.io/tree/main/examples/passport-example
